scripts/lint_jjb.py

```diff
@@ -87,7 +87,9 @@ def parse_jjb_file(in_dir, in_file):
 
 
 def parse_job_name(job_name, file_name):
-    regex = '^([a-zA-Z0-9]+-?)+([_]([{][a-zA-Z0-9_]+[}]-?)+)?$'
+    regex = (
+        '^([a-zA-Z0-9]+-?|\{[a-zA-Z0-9]+\}-?)+([_]([{][a-zA-Z0-9_]+[}]-?)+)?$'
```

Comment:
Looks ok:
![image](https://user-images.githubusercontent.com/253506/38747444-ccb50242-3f42-11e8-83b3-78d054eb6ebc.png)

Though the single character classes like `[_]` are a bit odd, though they are pre-existing.